File: src/main/kotlin/top/charjin/shoppingserver/service/OsOrderService.java
package top.charjin.shoppingserver.service;

import org.jetbrains.annotations.Nullable;
import org.springframework.stereotype.Service;
import top.charjin.shoppingserver.entity.OsOrder;
import top.charjin.shoppingserver.mapper.OsOrderMapper;
import top.charjin.shoppingserver.model.OsOrderModel;

import javax.annotation.Resource;
import java.util.List;

@Service
public class OsOrderService {

    @Resource
    private OsOrderMapper osOrderMapper;


    public int deleteByPrimaryKey(Integer orderId) {
        return osOrderMapper.deleteByPrimaryKey(orderId);
    }


    public int insert(OsOrder record) {
        return osOrderMapper.insert(record);
    }


    public int insertSelective(OsOrder record) {
        return osOrderMapper.insertSelective(record);
    }


    public OsOrder selectByPrimaryKey(Integer orderId) {
        return osOrderMapper.selectByPrimaryKey(orderId);
    }


    public int updateByPrimaryKeySelective(OsOrder record) {
        return osOrderMapper.updateByPrimaryKeySelective(record);
    }


    public int updateByPrimaryKey(OsOrder record) {
        return osOrderMapper.updateByPrimaryKey(record);
    }

    @Nullable
    public List<OsOrderModel> selectAllOrders(int user_id) {
        return osOrderMapper.selectAllOrders(user_id);
    }
}

File: src/main/kotlin/top/charjin/shoppingserver/service/OsOrderService.kt
package top.charjin.shoppingserver.service

import org.springframework.stereotype.Service
import top.charjin.shoppingserver.entity.OsOrder
import top.charjin.shoppingserver.mapper.OsOrderMapper
import top.charjin.shoppingserver.model.OsOrderModel

import javax.annotation.Resource

@Service
class OsOrderService {

    @Resource
    private lateinit var osOrderMapper: OsOrderMapper


    fun deleteByPrimaryKey(orderId: Int?): Int {
        return osOrderMapper.deleteByPrimaryKey(orderId)
    }


    fun insert(record: OsOrder): Int {
        return osOrderMapper.insert(record)
    }


    fun insertSelective(record: OsOrder): Int {
        return osOrderMapper.insertSelective(record)
    }


    fun selectByPrimaryKey(orderId: Int?): OsOrder {
        return osOrderMapper.selectByPrimaryKey(orderId)
    }


    fun updateByPrimaryKeySelective(record: OsOrder): Int {
        return osOrderMapper.updateByPrimaryKeySelective(record)
    }


    fun updateByPrimaryKey(record: OsOrder): Int {
        return osOrderMapper.updateByPrimaryKey(record)
    }

    fun selectAllOrders(user_id: Int): List<OsOrderModel>? {
        return osOrderMapper.selectAllOrders(user_id)
    }

    fun updateOrderStatusByOrderNo(orderNo: String, orderStatus: Int): Int {
        return osOrderMapper.updateOrderStatusByOrderNo(orderNo, orderStatus)
    }
}


